### PR TITLE
Fix for Github-949

### LINKED ISSUE #949: dependsOnMethods with alwaysRun = true and inheritance fails to find method
We have 2 classes: 
1. Parent class ClassA has methodA and methodB, methodA depends on methodB with alwaysRun = true. 
2. Child class: ClassB extends ClassA and overrides methodB.

When we run ClassB tests they do not run with the message:

> Method "ClassA.methodA()[pri:0, instance:ClassB@64cf5b3]" depends on nonexistent method "ClassA.methodB"

**Note:** this case was working in testng 4.6

### PR DESCRIPTION
**Pull request by krmahadevan**
dependsOnMethods with alwaysRun = true/false and 
inheritance fails to find method

Closes #949 #950 

Fixes #949  .

### Did you remember to?

- [X] Add test case(s)
- [X] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

File: src/test/java/test/inheritance/github949/ChildClassSample.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package test.inheritance.github949;
+
+import org.testng.annotations.Test;
+
+public class ChildClassSample extends ParentClassSample {
+    @Test
+    public void independent() {
```
Comment (juherr) [inline src/test/java/test/inheritance/github949/ChildClassSample.java]: You can add `@Override`

**Review by juherr — COMMENTED** [review]
LGTM :+1: 
Ping @cbeust

**Review by juherr — APPROVED** [review]

File: src/test/java/test/inheritance/github949/ChildClassSample.java (hunk 1 above)
Comment (krmahadevan) [reply src/test/java/test/inheritance/github949/ChildClassSample.java]: Fixed and forcibly pushed the changes so that its still a single commit.

Comment (juherr) [reply src/test/java/test/inheritance/github949/ChildClassSample.java]: :+1:

**Pull request merged**